### ISSUE TITLE
No proper documentation on how to integrate Cloud Reco in Native with Vuforia 10.

### ISSUE THREAD

**Issue by ti-dhavalprajapati** [PTCInc/vuforia-engine#32]
**Share which article or sample this is related to**
[A direct link to the article or description of the sample used.](https://library.vuforia.com/cloud-recognition/working-cloud-recognition#native)

**Elaborate on your feedback or issue**
There isn't much information about how one should integrate the code within their app. The article says "Create a Cloud Image Target Observer", but where should one supposed to put that code? Will it go into a funtion? Or in a whole new class? By leaving these kinds of information; devs ensures that the guidelines are followed and it helps with the code structure as well. They get more confidence on their code if the practices are suggested by the developers of the SDK/Library itself. Anyways, I tried implementing the code in the [sample project](https://developer.vuforia.com/downloads/samples) but got no luck. There are also some issues in the article like there's a type in "Handling Search Results" section where the V is capitalized in 'Void'. In the same "recoEventHandler" function; variable 'mEngine' is used, despite it is being private in the AppController.h. There is no sample project for the Cloud Reco feature, which is something should be prioritized in order to get people to use the SDK.

**Describe the solution you'd like**
 - Update article containing proper guidelines/practices and leave more info on how one should integrate the feature STEP BY STEP. Read the article/documentation 5 to 10 times and make sure there are no typos/errors.
 - Include sample project so everyone can follow the structure and the guidelines.

**Describe possible alternatives you've considered**

**Additional context**

**Comment by siddhantjain4R** [comment]
yes facing the same issue when trying to integrate vuforia for iOS. Documentation is not clear.

**Comment by ptc-thesbyro** [comment]
Hello, and thank you for bringing this up.
We can correct the typos and clarify the current setup in more detail. But as you mention, that is not enough to get the feature working in your project. To better identify where information is lacking, could you describe your approach to adding cloud image recognition to the native sample project?

**Comment by ti-dhavalprajapati** [comment]
Thanks for the reply. Luckily I was able to implement cloud recognition into my project. I had to do some changes in order to get it working. The `mEngine` variable is private inside the header file. So as provided in the `recoEventHandler` method, it shouldn't work because of the scope of the variable. One way to access it is to make the method `AppController::recoEventHandler`. Doing so also won't work because then `vuCloudImageTargetObserverRegisterHandlers` will not accept that as a parameter. Then I created a new local `mEngine1` varialbe inside cpp file and assigned the actual `mEngine` variable when the vuforia successfully starts/initializes.

Now that's all done. Now to access all these inside Swift or Kotlin, I will also have to do some work and create a wrapper to use it inside project. IMHO all these should be included in the SDK itself. My next suggestion is to update the article, code and provide an example for the community. I believe that experienced devs might be able to crack it and implement it, but what has been provided right now isn't friendly enough for new developers.

**Comment by ptc-thesbyro** [comment]
Thanks for the additional information on your approach and remaining implementation plans.
I will get back to you with an updated guide and recommendations with steps on how to achieve Cloud Image recognition in a native project.

**Comment by ti-dhavalprajapati** [comment]
Thank you very much. Looking forward to an update.

**Comment by ti-dhavalprajapati** [comment]
![image](https://github.com/PTCInc/vuforia-engine/assets/69501026/33f053a0-6744-4424-b394-2566bb4aab3a)

Just wanted to point out that above image contains `cloudImageTargetFeature` from the `recoEventHandler` method that I searched in the [C API Reference](https://library.vuforia.com/sites/default/files/references/native/index.html), but it doesn't exist. Not sure if it is important or not. If there is something supposed to be there, then please tell me as I'm working continously to integrate Cloud Reco in my project.

**Comment by ti-dhavalprajapati** [comment]
`ninja: error: 'D:/User/Projects/Android Native/vuforia-sdk-android-10-15-3/samples/vuforia-sample-10-15-3/Android/app/build/arcore-native/jni/arm64-v8a/libarcore_sdk_c.so', needed by 'D:/User/Projects/Android Native/vuforia-sdk-android-10-15-3/samples/vuforia-sample-10-15-3/Android/app/build/intermediates/cxx/Debug/xe1t323i/obj/arm64-v8a/libVuforiaSample.so', missing and no known rule to make it`

Integrated and tried to run Cloud Reco project in android and got above error. Don't know how to resolve it, the sample provided by vuforia works just fine. In iOS Cloud Reco is working properly. Not sure if I have missed something while integrating.

It'd be great if you can just provide a working example of Cloud Reco for Android. Thank you.

**Comment by ti-dhavalprajapati** [comment]
Hi @ptc-thesbyro,

Would love to get an update on this. As I mentioned we are continuously trying to get this working.

**Comment by Brianheimer** [comment]
I too would like an update to this issue.  Is it currently being worked on?

**Comment by ptc-thesbyro** [comment]
Hello, 

I can confirm that the issue has been raised and is being worked on. Unfortunately, I cannot give you an estimate for when the improved documentation will be ready.

**Comment by ti-dhavalprajapati** [comment]
Hi @ptc-thesbyro,

Since its been almost a month. I'd love to have an update on this.

**Comment by ptc-thesbyro** [comment]
Apologies for the long wait. 

I have published an improved guide for cloud recognition: [Cloud Recognition API Overview](https://library.vuforia.com/cloud-recognition/cloud-recognition-api-overview)

With it, I have corrected the reported errors and I have added code snippets that can be copied more or less as is.

Thank you for your patience. Let me know if you have any other questions or feedback based on the new material.

**Comment by ptc-pscheper** [comment]
This issue is closed due to inactivity. Feel free to re-open or create a new bug report in the future.
